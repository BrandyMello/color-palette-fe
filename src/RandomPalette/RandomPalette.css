h3 {
  font-size: 30px;
  text-align: center;
}

.swatch {
  border: 1px solid black;
  border-radius: 8px;
  display: inline-block;
  height: 300px;
  width: 19.8%;
}

.generate_palette_button {
  background-color: #808080;
  border: 1px solid black;
  border-radius: 3px;
  color: white;
  font-size: 30px;
  font-weight: bold;
  height: 50px;
  width: 600px;
  margin: 5px;
  margin-left: 31%;
}

.generate_palette_button:hover {
  background-color: #46a3e4;
}

.palette-section {
  border: 2px solid #808080;
  border-radius: 8px;
  margin-top: 30px;
  padding-bottom: 30px;
}