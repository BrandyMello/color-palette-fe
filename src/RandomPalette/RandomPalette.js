import React from 'react'
import './RandomPalette.css'

const RandomPalette = ({ colors, generateRandomPalette }) => {
  const hexColor = colors.map((color, index) => {
    return (
      <div 
      className ="swatch"
      style={{backgroundColor: color[`color_${index+1}`]}}
      >
      </div>
    )
  })
  return (
    <div>
      { hexColor }
      <button type="submit" onClick={() => generateRandomPalette()}>Generate Palette</button>
    </div>
  )
}

export default RandomPalette