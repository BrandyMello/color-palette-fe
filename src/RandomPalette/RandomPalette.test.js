import React from 'react';
import { shallow } from 'enzyme';
import RandomPalette from './RandomPalette';

describe('RandomPalette', () => {
  let wrapper;
  let mockPalette;

  beforeEach(() => {
    mockPalette = [
      {
        name: "SeaBreeze",
        projectName: "Cool Colors",
        colorOne: "#013163",
        colorTwo: "#015E7E",
        colorThree: "#00AAB2",
        colorFour: "#C5BBCC",
        colorFive: "#A96B70",
        projectId: 1
      }
    ];
    wrapper = shallow(<RandomPalette colors={mockPalette} generateRandomPalette={jest.fn()} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});