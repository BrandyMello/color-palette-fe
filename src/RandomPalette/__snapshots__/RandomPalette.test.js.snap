// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomPalette should match snapshot 1`] = `
<div
  className="button_div"
>
  <div
    className="swatch"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />
  <button
    className="generate_palette_button"
    onClick={[Function]}
    type="submit"
  >
    Generate Palette
  </button>
</div>
`;
